### ISSUE TITLE
jetson agx orin 上安装apollo错误

### ISSUE THREAD

**Issue by 1446311191** [ApolloAuto/apollo#14505]
### 系统信息

Package: nvidia-jetpack
Version: 5.0.1-b118
Architecture: arm64
Maintainer: NVIDIA Corporation
Installed-Size: 194
Depends: nvidia-cuda (= 5.0.1-b118), nvidia-opencv (= 5.0.1-b118), nvidia-cudnn8 (= 5.0.1-b118), nvidia-tensorrt (= 5.0.1-b118), nvidia-container (= 5.0.1-b118), nvidia-vpi (= 5.0.1-b118), nvidia-nsight-sys (= 5.0.1-b118), nvidia-l4t-jetson-multimedia-api (>> 34.1-0), nvidia-l4t-jetson-multimedia-api (<< 34.2-0)
Homepage: http://developer.nvidia.com/jetson
Priority: standard
Section: metapackages
Filename: pool/main/n/nvidia-jetpack/nvidia-jetpack_5.0.1-b118_arm64.deb
Size: 29356
SHA256: 379f1b0c833a7222c37793b766b3b278645e791859c2aa33a89fd3450a45f872
SHA1: 1c409cd2a0f0682f894d82be707e60b248981893
MD5sum: d9cfb50922e61be108889e2bf6168970
Description: NVIDIA Jetpack Meta Package
Description-md5: ad1462289bdbc54909ae109d1d32c0a8

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.4 LTS
Release:        20.04
Codename:       focal

Linux ubuntu 5.10.65-tegra #1 SMP PREEMPT Mon May 16 20:58:07 PDT 2022 aarch64 aarch64 aarch64 GNU/Linux

问题：
执行sudo bash docker/scripts/dev_start.sh时出现错误

> sudo bash docker/scripts/dev_start.sh 
[INFO] Use default GeoLocation settings
[INFO] Start pulling docker image apolloauto/apollo:dev-aarch64-18.04-20201218_0030 ...
dev-aarch64-18.04-20201218_0030: Pulling from apolloauto/apollo
Digest: sha256:9d265275fd175be3de5f803cfbe1ccca74aab34231d61b4ca5c4ce391755834c
Status: Image is up to date for apolloauto/apollo:dev-aarch64-18.04-20201218_0030
docker.io/apolloauto/apollo:dev-aarch64-18.04-20201218_0030
[INFO] Remove existing Apollo Development container ...
[INFO] Removing existing Apollo container: apollo_dev_root
apollo_dev_root
[INFO] Determine whether host GPU is available ...
[INFO] USE_GPU_HOST: 1
[WARNING] No CAN device named /dev/can0. 
[WARNING] No SocketCAN device named can0.
[INFO] Starting mounting map volumes ...
[INFO] Load map sunnyvale_big_loop from image: apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
[INFO] Create volume apollo_map_volume-sunnyvale_big_loop_root from image: apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest ...
map_volume-sunnyvale_big_loop-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:275f66d9200b1e947ba27ae7e78ee0cdc455203d9f1e35e3f32b02e63fa5361e
Status: Image is up to date for apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
docker.io/apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
[INFO] Load map sunnyvale_loop from image: apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
[INFO] Create volume apollo_map_volume-sunnyvale_loop_root from image: apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest ...
map_volume-sunnyvale_loop-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:fe449650dd5634b7c4ed29255d2537a41a34f7f2551cc1bb18013deeb70da447
Status: Image is up to date for apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
docker.io/apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
[INFO] Load map sunnyvale_with_two_offices from image: apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest
[INFO] Create volume apollo_map_volume-sunnyvale_with_two_offices_root from image: apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest ...
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 127.0.0.53:53: read udp 127.0.0.1:60470->127.0.0.53:53: i/o timeout
[ERROR] Failed to pull docker image : apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest

> sudo bash docker/scripts/dev_start.sh 
[INFO] Use default GeoLocation settings
[INFO] Start pulling docker image apolloauto/apollo:dev-aarch64-18.04-20201218_0030 ...
dev-aarch64-18.04-20201218_0030: Pulling from apolloauto/apollo
Digest: sha256:9d265275fd175be3de5f803cfbe1ccca74aab34231d61b4ca5c4ce391755834c
Status: Image is up to date for apolloauto/apollo:dev-aarch64-18.04-20201218_0030
docker.io/apolloauto/apollo:dev-aarch64-18.04-20201218_0030
[INFO] Remove existing Apollo Development container ...
[INFO] Determine whether host GPU is available ...
[INFO] USE_GPU_HOST: 1
[WARNING] No CAN device named /dev/can0. 
[WARNING] No SocketCAN device named can0.
[INFO] Starting mounting map volumes ...
[INFO] Load map sunnyvale_big_loop from image: apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
[INFO] Create volume apollo_map_volume-sunnyvale_big_loop_root from image: apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest ...
map_volume-sunnyvale_big_loop-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:275f66d9200b1e947ba27ae7e78ee0cdc455203d9f1e35e3f32b02e63fa5361e
Status: Image is up to date for apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
docker.io/apolloauto/apollo:map_volume-sunnyvale_big_loop-aarch64-latest
[INFO] Load map sunnyvale_loop from image: apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
[INFO] Create volume apollo_map_volume-sunnyvale_loop_root from image: apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest ...
map_volume-sunnyvale_loop-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:fe449650dd5634b7c4ed29255d2537a41a34f7f2551cc1bb18013deeb70da447
Status: Image is up to date for apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
docker.io/apolloauto/apollo:map_volume-sunnyvale_loop-aarch64-latest
[INFO] Load map sunnyvale_with_two_offices from image: apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest
[INFO] Create volume apollo_map_volume-sunnyvale_with_two_offices_root from image: apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest ...
map_volume-sunnyvale_with_two_offices-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:13ae92aac07a2aeca873356cf525c26e0e316d642ad6f811cc15b4aa2946ebc7
Status: Image is up to date for apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest
docker.io/apolloauto/apollo:map_volume-sunnyvale_with_two_offices-aarch64-latest
[INFO] Load map san_mateo from image: apolloauto/apollo:map_volume-san_mateo-aarch64-latest
[INFO] Create volume apollo_map_volume-san_mateo_root from image: apolloauto/apollo:map_volume-san_mateo-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:map_volume-san_mateo-aarch64-latest ...
map_volume-san_mateo-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:23de05457d602e35605ebbefba5f26f18b50c79502072f44af2b605a0c249d5a
Status: Image is up to date for apolloauto/apollo:map_volume-san_mateo-aarch64-latest
docker.io/apolloauto/apollo:map_volume-san_mateo-aarch64-latest
[INFO] Mount other volumes ...
[INFO] Create volume apollo_audio_volume_root from image: apolloauto/apollo:data_volume-audio_model-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:data_volume-audio_model-aarch64-latest ...
data_volume-audio_model-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:5bb275a29aac2147cbddf943b19666c62ba5bd1ebd4695f8e5f54da83802945d
Status: Image is up to date for apolloauto/apollo:data_volume-audio_model-aarch64-latest
docker.io/apolloauto/apollo:data_volume-audio_model-aarch64-latest
[INFO] Create volume apollo_yolov4_volume_root from image: apolloauto/apollo:yolov4_volume-emergency_detection_model-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:yolov4_volume-emergency_detection_model-aarch64-latest ...
yolov4_volume-emergency_detection_model-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:4883d56073054bd8170a66dbd519fc7238bdde80eb436974710803d484bdc73b
Status: Image is up to date for apolloauto/apollo:yolov4_volume-emergency_detection_model-aarch64-latest
docker.io/apolloauto/apollo:yolov4_volume-emergency_detection_model-aarch64-latest
[INFO] Create volume apollo_faster_rcnn_volume_root from image: apolloauto/apollo:faster_rcnn_volume-traffic_light_detection_model-aarch64-latest
[INFO] Start pulling docker image apolloauto/apollo:faster_rcnn_volume-traffic_light_detection_model-aarch64-latest ...
faster_rcnn_volume-traffic_light_detection_model-aarch64-latest: Pulling from apolloauto/apollo
Digest: sha256:2ad8bccebec1cd91620b7179a15c23f20bec90faddc12cf98f9cc1e0a515d96e
Status: Image is up to date for apolloauto/apollo:faster_rcnn_volume-traffic_light_detection_model-aarch64-latest
docker.io/apolloauto/apollo:faster_rcnn_volume-traffic_light_detection_model-aarch64-latest
[INFO] Starting Docker container "apollo_dev_root" ...
+ docker run --gpus all -itd --privileged --name apollo_dev_root -e DISPLAY=:0 -e DOCKER_USER=root -e USER=root -e DOCKER_USER_ID=0 -e DOCKER_GRP=root -e DOCKER_GRP_ID=0 -e DOCKER_IMG=apolloauto/apollo:dev-aarch64-18.04-20201218_0030 -e USE_GPU_HOST=1 -e NVIDIA_VISIBLE_DEVICES=all -e NVIDIA_DRIVER_CAPABILITIES=compute,video,graphics,utility --volume apollo_map_volume-sunnyvale_big_loop_root:/apollo/modules/map/data/sunnyvale_big_loop --volume apollo_map_volume-sunnyvale_loop_root:/apollo/modules/map/data/sunnyvale_loop --volume apollo_map_volume-sunnyvale_with_two_offices_root:/apollo/modules/map/data/sunnyvale_with_two_offices --volume apollo_map_volume-san_mateo_root:/apollo/modules/map/data/san_mateo --volume apollo_audio_volume_root:/apollo/modules/audio/data/ --volume apollo_yolov4_volume_root:/apollo/modules/perception/camera/lib/obstacle/detector/yolov4/model/ --volume apollo_faster_rcnn_volume_root:/apollo/modules/perception/production/data/perception/camera/models/traffic_light_detection/faster_rcnn_model -v /home/zhou/github/apollo:/apollo -v /dev:/dev -v /media:/media -v /tmp/.X11-unix:/tmp/.X11-unix:rw -v /etc/localtime:/etc/localtime:ro -v /usr/src:/usr/src -v /lib/modules:/lib/modules --net host -w /apollo --add-host in-dev-docker:127.0.0.1 --add-host ubuntu:127.0.0.1 --hostname in-dev-docker --shm-size 2G --pid=host -v /dev/null:/dev/raw1394 apolloauto/apollo:dev-aarch64-18.04-20201218_0030 /bin/bash
e5e82f5279c08740dafcdaa12bb1e34d5fbac40e75f0584080d45022d5708c5a
docker: Error response from daemon: failed to create shim: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: Auto-detected mode as 'csv'
invoking the NVIDIA Container Runtime Hook directly (e.g. specifying the docker --gpus flag) is not supported. Please use the NVIDIA Container Runtime instead.: unknown.
+ '[' 125 -ne 0 ']'
+ error 'Failed to start docker container "apollo_dev_root" based on image: apolloauto/apollo:dev-aarch64-18.04-20201218_0030'
+ echo -e '[\033[0;31mERROR\033[0m] Failed to start docker container "apollo_dev_root" based on image: apolloauto/apollo:dev-aarch64-18.04-20201218_0030'
[ERROR] Failed to start docker container "apollo_dev_root" based on image: apolloauto/apollo:dev-aarch64-18.04-20201218_0030
+ exit 1

**Comment by roryleo** [comment]
aarch64 docker image don't support jetson orin now.

**Comment by daohu527** [comment]
The first error is cause by pull docker failed, and then we focused on the second, since you just run the start docker cmd, so the problem will just be limited to **how to start the arm version of navidia docker**, the compilation and running problems are still behind.

first I think you should check nvidia docker is right, use `nvidia-smi`, and I search some problem about the error, there may be many reasons, so we will do step by step！

**Comment by SNC6SI** [comment]
> The first error is cause by pull docker failed, and then we focused on the second, since you just run the start docker cmd, so the problem will just be limited to **how to start the arm version of navidia docker**, the compilation and running problems are still behind.
> 
> first I think you should check nvidia docker is right, use `nvidia-smi`, and I search some problem about the error, there may be many reasons, so we will do step by step！ ref [link](https://github.com/NVIDIA/nvidia-docker/issues/1648)

maybe there is no `nvidia-smi` on platform  @aarch64?

**Comment by daohu527** [comment]
I check that Jetson platform not support `nvidia-smi`. ref [link](https://forums.developer.nvidia.com/t/how-to-watch-the-memory-on-an-arm-development-board-with-a-jetson-tx2-gpu/53679)

**Comment by daohu527** [comment]
Jetson‘s support will be a future work, If there is an update, we will notify you as soon as possible.

**Comment by QiTianDaShengMaoHuiFei** [comment]
我也遇到了同样的问题，请问你是如何解决的？

**Comment by SniperZhao** [comment]
any updates?

**Comment by KelvinJiang** [comment]
I also encountered this issue when using Jetson Orin Nano. I found that the dev_start.sh script in Apollo is judged as aarch64, and when `lsmod | grep nvgpu ` returns ok, it will use `docker run --gpus all ` to start the docker container. I manually tried to run the official image with the command `docker run --gpus all nvidia/cuda:11.2.0-runtime-ubuntu20.04 nvidia-smi` , but it failed . The reason is that the nvidia driver cannot be mounted into the docker.

I have installed nvidia-container-runtime in my environment and updated /etc/docker/daemon.json:
`
{
"runtimes": {
"nvidia": {
"args": [],
"path": "nvidia-container-runtime"
}
}
}
`
However, I still cannot use `docker run --gpus all`  to start the container. It seems that nvidia-smi cannot be used on Orin Nano to check the graphics card information, so I cannot confirm whether the Nvidia driver has been successfully installed.
